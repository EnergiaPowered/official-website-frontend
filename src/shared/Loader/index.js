import React from 'react';
/**
 * Component which show untill the data get to the user
 * 
 * @component 
 * @returns {JSX} Loadding Shap
 */
export default () => (
    <div className="col-12 d-flex flex-column align-items-center p-5">
        <span className="fa fa-spinner fa-pulse fa-3x fa-fw text-primary" style={{ fontSize: "24px" }}></span>
        <p className="pt-3" style={{ fontSize: "24px" }}>Loading...</p>
    </div>
);