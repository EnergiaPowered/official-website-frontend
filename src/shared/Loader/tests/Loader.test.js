import React from "react";
import ReactDOM from "react-dom";
import Loader from "./../../Loader/index";
import { cleanup } from "@testing-library/react";
import { create } from "react-test-renderer";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

// test for loader
afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Loader />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
    const loader = create(<Loader />).toJSON();
    expect(loader).toMatchSnapshot();
});
Enzyme.configure({ adapter: new Adapter() });

it("check default title", () => {
    const loader = mount(<Loader />);
    expect(loader.find("p").text()).toEqual("Loading...");
    expect(loader.toJSON).toMatchSnapshot();
});
