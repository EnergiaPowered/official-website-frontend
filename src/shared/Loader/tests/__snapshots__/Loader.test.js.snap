// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check default title 1`] = `undefined`;

exports[`matches snapshot 1`] = `
<div
  className="col-12 d-flex flex-column align-items-center p-5"
>
  <span
    className="fa fa-spinner fa-pulse fa-3x fa-fw text-primary"
    style={
      Object {
        "fontSize": "24px",
      }
    }
  />
  <p
    className="pt-3"
    style={
      Object {
        "fontSize": "24px",
      }
    }
  >
    Loading...
  </p>
</div>
`;
